### ISSUE TITLE
Handle generic type with trait bound `Future`

### ISSUE THREAD

**Issue by ManevilleF** [fMeow/maybe-async-rs#4]
Hello,

I use the `maybe_async` crate for a few projects, and it works great. I only have a issue on the tests:

This is a test from my crate [aragog](https://crates.io/crates/aragog) based on your `arangors` crate.

```rust
#[maybe_async::test(
        feature = "blocking",
        async(all(not(feature = "blocking")), tokio::test)
    )]
    async fn can_be_recorded_and_retrieved() -> Result<(), String> {
        with_db(|pool| async move {
            let dish = init_dish();
            let dish_record = DatabaseRecord::create(dish, &pool).await.unwrap();
            let found_record = Dish::find(&dish_record.key, &pool).await.unwrap();
            common::expect_assert_eq(dish_record.record, found_record.record)?;
            Ok(())
        })
        .await
    }
```

When `async` it works fine, but when I use the `blocking` feature the `async move` is not changed making the test build fail.
I'm wondering if handling this case could be done through `mabe_async`

Otherwise I can chang the way I write my db-related tests.

EDIT:

the `with_db`function is the following:
```rust
#[maybe_async::maybe_async]
pub async fn with_db<T, F>(test: T) -> Result<(), String>
where
    T: FnOnce(DatabaseConnectionPool) -> F,
    F: Future<Output = Result<(), String>>,
{
    let db_pool = setup_db().await;
    test(db_pool).await
}
```

Mybe the issue is comming for the `F` bound to a `Future`

**Comment by fMeow** [comment]
Sorry for the long wait.

This is definitely a bug and I will look into this issue in a few days.

**Comment by fMeow** [comment]
This error comes from the function signatures `F: Future<Output = Result<(), String>>`. `maybe-async` currently simply remove all async and await keyword. I think a quick solution is to replace all `F: Future<Output = AnyType>` to `AnyType`.

**Comment by fMeow** [comment]
This issue is solved by release 0.2.4. Though the implementation is a little error prone.

It currently search generic type with trait bound that ends with `Future` in both generic argument declaration and where cluase, and then replace the corresponding generic type with its Output type. 

So if a) another crate other than `futures` named a trait  `Future` or b) `Future` is renamed, the current implementation will fail to remove this trait bound.
